src/App.tsx

```diff
@@ -12,13 +19,120 @@ const goodsFromServer: string[] = [
   'Honey',
   'Jam',
   'Garlic',
-];
-
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+].map(good => ({
+  id: uuidv4(),
+  value: good,
+}));
+
+type Props = {};
+
+interface State {
+  selectedGoods: string[],
+}
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearSelection = () => this.setState({ selectedGoods: [] });
+
+  addGood = (goodToAdd: string) => {
+    const { selectedGoods } = this.state;
+
+    this.setState({ selectedGoods: [...selectedGoods, goodToAdd] });
+  };
+
+  removeGood = (goodToRemove: string) => {
+    const { selectedGoods } = this.state;
+
+    this.setState({
+      selectedGoods: selectedGoods.filter(
+        good => good !== goodToRemove,
+      ),
+    });
+  };
+
+  formatHeading = () => {
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case 0:
+        return 'No goods selected';
+
+      case 1:
+        return `${selectedGoods[0]} is selected`;
+
+      default:
+        return `${selectedGoods.slice(0, -1).join(', ')}
+         and ${selectedGoods[selectedGoods.length - 1]}
```

Comment:
```suggestion
         and ${selectedgoods.slice(-1)} 
```
Чтоб укоротить запись я просто писала так